client/rest/test/plugins/metadata/metadataRoutes_spec.js

```diff
@@ -240,4 +240,45 @@ describe('metadata routes', () => {
 			});
 		});
 	});
+	describe('metal', () => {
+		const mockServer = new MockServer();
+		const db = {
+			binDataByMetalId: sinon.stub().resolves({ payload: 'db_payload', text: 'db_text' })
+		};
+		metadataRoutes.register(mockServer.server, db, {});
+
+		beforeEach(() => {
+			mockServer.resetStats();
+			mockServer.res.setHeader = sinon.spy();
+			mockServer.res.write = sinon.spy();
+			mockServer.res.end = sinon.spy();
+		});
+
+		describe('get by metal id', () => {
+			it('returns page with results', () => {
+				// Arrange:
+				const route = mockServer.getRoute('/metadata/metal/:metalId').get();
+				const metalId = 'your_metal_id';
+				const mimeType = 'image/png';
+				const fileName = 'image.png';
+				const download = 'true';
+				const req = {
+					params: {
+						metalId, mimeType, fileName, download
+					}
+				};
+				// Act:
+				return mockServer.callRoute(route, req).then(() => {
+					// Assert:
+					expect(mockServer.res.write.calledOnce).to.equal(true);
+					expect(mockServer.next.calledOnce).to.equal(true);
+					expect(mockServer.res.setHeader.calledWithExactly('content-type', 'image/png')).to.equal(true);
+					expect(mockServer.res.setHeader
+						.calledWithExactly('Content-Disposition', 'attachment; filename="image.png"')).to.equal(true);
+					expect(db.binDataByMetalId.calledOnce).to.equal(true);
+					expect(db.binDataByMetalId.alwaysCalledWith(metalId)).to.equal(true);
```

Comment:
typically, routeUtils contains implementation details, but routes are directly interacted with by users.

it is generally recommended to check all the things that users may rely on at this level.
since `Content-MetalText` is potentially something that is important to a user, we should check that it is set correctly (or is not present if not available).

Reply:
sorry, I was completely mistaken🙏
```js
await mockServer.callRoute(route, req).then(() => {
	expect(mockServer.res.write.calledOnce).to.equal(true);
	expect(mockServer.next.calledOnce).to.equal(true);
	expect(mockServer.res.setHeader.calledWithExactly('content-type', 'image/png')).to.equal(true);
	expect(mockServer.res.setHeader
		.calledWithExactly('Content-Disposition', 'attachment; filename="image.png"')).to.equal(true);
	expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_text')).to.equal(true); // i added
	expect(db.binDataByMetalId.calledOnce).to.equal(true);
	expect(db.binDataByMetalId.alwaysCalledWith('metal_id')).to.equal(true);
});
```